drivers/media/i2c/imx477.c

```diff
@@ -1285,20 +1292,49 @@ static void imx477_adjust_exposure_range(struct imx477 *imx477,
 
 	/* Honour the VBLANK limits when setting exposure. */
 	exposure_max = imx477->mode->height + imx477->vblank->val -
-						IMX477_EXPOSURE_OFFSET;
+		       (IMX477_EXPOSURE_OFFSET << imx477->long_exp_shift);
 	exposure_def = min(exposure_max, imx477->exposure->val);
 	__v4l2_ctrl_modify_range(imx477->exposure, imx477->exposure->minimum,
 				 exposure_max, imx477->exposure->step,
 				 exposure_def);
 }
 
+static int imx477_set_frame_length(struct imx477 *imx477, unsigned int val)
+{
+	int ret = 0;
+
+	imx477->long_exp_shift = 0;
+
+	while (val > IMX477_FRAME_LENGTH_MAX) {
+		if (++imx477->long_exp_shift > IMX477_LONG_EXP_SHIFT_MAX) {
```

Comment:
Pre/post increments/decrements in logical statements always make me have to think harder than I'd like over the actual logic being checked. Separate them into two lines?